pycode/memilio-surrogatemodel/memilio/surrogatemodel/ode_secir_groups/data_generation.py

```diff
@@ -0,0 +1,302 @@
+    result_array = result.as_ndarray()
+
+    # Omit first column, as the time points are not of interest here.
+    dataset_entries = copy.deepcopy(result_array[1:, :].transpose())
+
+    return dataset_entries.tolist(), damped_contact_matrix
+
+
+def generate_data(
+        num_runs, path_out, path_population, input_width, label_width,
+        normalize=True, save_data=True):
+    """! Generate data sets of num_runs many equation-based model simulations and transforms the computed results by a log(1+x) transformation.
+    Divides the results in input and label data sets and returns them as a dictionary of two TensorFlow Stacks.
+    In general, we have 8 different compartments and 6 age groups.  If we choose, 
+    input_width = 5 and label_width = 20, the dataset has 
+    - input with dimension 5 x 8 x 6
+    - labels with dimension 20 x 8 x 6
+   @param num_runs Number of times, the function run_secir_groups_simulation is called.
+   @param path_out Path, where the dataset is saved to.
+   @param path_population Path, where we try to read the population data.
+   @param input_width Int value that defines the number of time series used for the input.
+   @param label_width Int value that defines the size of the labels.
+   @param normalize [Default: true] Option to transform dataset by logarithmic normalization.
+   @param save_data [Default: true] Option to save the dataset.
+   @return Data dictionary of input and label data sets.
+   """
+    data = {
+        "inputs": [],
+        "labels": [],
+        "contact_matrix": [],
+        "damping_day": []
+    }
+
+    # The number of days is the same as the sum of input and label width.
+    # Since the first day of the input is day 0, we still need to subtract 1.
+    days = input_width + label_width - 1
+
+    # Load population data
+    population = get_population(path_population)
+
+    # show progess in terminal for longer runs
+    # Due to the random structure, theres currently no need to shuffle the data
+    bar = Bar('Number of Runs done', max=num_runs)
+    for _ in range(0, num_runs):
+
+        # Generate a random damping day
+        damping_day = random.randrange(
+            input_width, input_width+label_width)
+
+        data_run, damped_contact_matrix = run_secir_groups_simulation(
+            days, damping_day, population[random.randint(0, len(population) - 1)])
+        data['inputs'].append(data_run[:input_width])
+        data['labels'].append(data_run[input_width:])
+        data['contact_matrix'].append(np.array(damped_contact_matrix))
+        data['damping_day'].append(damping_day)
+        bar.next()
+    bar.finish()
+
+    if normalize:
+        # logarithmic normalization
+        transformer = FunctionTransformer(np.log1p, validate=True)
+        inputs = np.asarray(data['inputs']).transpose(2, 0, 1).reshape(48, -1)
+        scaled_inputs = transformer.transform(inputs)
+        scaled_inputs = scaled_inputs.transpose().reshape(num_runs, input_width, 48)
```

Comment:
input dimension is not input_width x num_compartments x num_age_groups any more but is now input_width x 48